Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/CruiseControl.cs

```diff
@@ -0,0 +1,2277 @@
+                    foreach (TrainCar tc in Locomotive.Train.Cars)
+                    {
+                        if (tc.GetType() == typeof(MSTSLocomotive) || tc.GetType() == typeof(MSTSDieselLocomotive) || tc.GetType() == typeof(MSTSElectricLocomotive))
+                        {
+                            if (tc != Locomotive)
+                            {
+                                try
+                                {
+                                    playerNotDriveableTrainLocomotives.Add((MSTSLocomotive)tc);
+                                }
+                                catch { }
+                            }
+                        }
+                    }
+                    firstIteration = false;
+                }
+
+                if (SelectedMaxAccelerationStep == 0) // no effort, no throttle (i.e. for reverser change, etc) and return
+                {
+                    Locomotive.SetThrottlePercent(0);
+                    if (Locomotive.DynamicBrakePercent > 0)
+                        Locomotive.SetDynamicBrakePercent(0);
+                }
+
+                if (SpeedRegMode == SpeedRegulatorMode.Auto)
+                {
+                    if (SpeedSelMode == SpeedSelectorMode.Parking && !Locomotive.EngineBrakePriority)
+                    {
+                        if (Locomotive.DynamicBrakePercent > 0)
+                        {
+                            if (AbsWheelSpeedMpS == 0)
+                            {
+                                Locomotive.SetDynamicBrakePercent(0);
+                                Locomotive.DynamicBrakeChangeActiveState(false);
+                            }
+                        }
+                        if (!UseThrottle) Locomotive.ThrottleController.SetPercent(0);
+                        throttleIsZero = true;
+
+                        if (AbsWheelSpeedMpS < (SpeedIsMph ? MpS.FromMpH(ParkingBrakeEngageSpeed) : MpS.FromKpH(ParkingBrakeEngageSpeed)))
+                            Locomotive.SetEngineBrakePercent(ParkingBrakePercent);
+                    }
+                    else if (SpeedSelMode == SpeedSelectorMode.Neutral || SpeedSelMode < SpeedSelectorMode.Start && !SpeedRegulatorOptions.Contains("startfromzero") && AbsWheelSpeedMpS < SafeSpeedForAutomaticOperationMpS)
+                    {
+                        if (controllerVolts > 0)
+                        {
+                            float step = 100 / ThrottleFullRangeIncreaseTimeSeconds;
+                            step *= elapsedClockSeconds;
+                            controllerVolts -= step;
+                            if (controllerVolts < 0) controllerVolts = 0;
+                            if (controllerVolts > 0 && controllerVolts < 0.1) controllerVolts = 0;
+                        }
+
+                        float delta = 0;
```

Comment:
Probably worth it to include units (i.e. deltaSpeedMpS)

Reply:
Done